jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java

```diff
@@ -302,6 +304,21 @@ public CRIUSupport setUnprivileged(boolean unprivileged) {
 		return this;
 	}
 
+	/**
+	 * Set the size limit for ghost files when taking a checkpoint. File limit
+	 * can not be greater than 2^32 - 1 or negative.
+	 *
+	 * Default: 1MB set by CRIU
+	 *
+	 * @param limit the file limit size in bytes.
+	 * @return this
+	 * @throws UnsupportedOperationException if file limit is greater than 2^32 - 1 or negative.
+	 */
+	public CRIUSupport setGhostFileLimit(long limit) {
+		internalCRIUSupport = internalCRIUSupport.setGhostFileLimit(limit);
```

Comment:
Are there APIs in the internal which require creating a new instance? Could lead to problems if calls are made from multiple threads.

Reply:
Its a known limiation that builder pattern is susceptible to thread races if constructed in a multi-threaded fashion. Its standard practice that the consumer must ensure thread safety.